src/core.js

```diff
@@ -1071,7 +1058,7 @@ define([
                 _this.data.core.formLoadingFailed = true;
                 _this.data.core.failedLoadXML = formString;
 
-                $.fancybox.hideActivity();
+                _this.closeModal();
```

Comment:
Indeed, it will. Fixed.

I agree that the modal code could use refactoring. Any idea why we currently have an fd-modal-container and then another container for multimedia, with one modal for each media type inside that? I'd think we'd be fine with only ever having one modal on the page, appending it to the body (or $f), and having it remove itself when hidden (with the code that created it also responsible for removing it).